backend/btrixcloud/operator.py

```diff
@@ -447,20 +478,29 @@ async def inc_crawl_complete_stats(self, crawl, finished):
     async def add_crawl_errors_to_db(self, redis, crawl_id, inc=100):
         """Pull crawl errors from redis and write to mongo db"""
         index = 0
-        while True:
-            skip = index * inc
-            upper_bound = skip + inc - 1
-            errors = await redis.lrange(f"{crawl_id}:e", skip, upper_bound)
-            if not errors:
-                break
-
-            await add_crawl_errors(self.crawls, crawl_id, errors)
-
-            if len(errors) < inc:
-                # If we have fewer than inc errors, we can assume this is the
-                # last page of data to add.
-                break
-            index += 1
+        try:
+            # ensure this only runs once
+            if not await redis.setnx("errors-exported", "1"):
+                return
```

Comment:
clever!

Reply:
yeah, realized that it could run more than once, as everything in operator is supposed to be idempotent, so should prevent duplicate errors being added.